File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
    int n = nums.size();
    unordered_map<int, bool> isExist;

    for (int i = 0; i < n; i++) {
        if (isExist[nums[i]]) {
            nums.erase(nums.begin() + i);
            i--;
            n--;
        }
        else {
            isExist[nums[i]] = true;
        }
    }

    return nums.size();
}
};

File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int n = nums.length;
        int left = 1;
        int res = 1;
        for(int i = 1 ; i < n ; i++){
            if(nums[i] != nums[i - 1]){
                res++;
                nums[left] = nums[i];
                left++;
            }
        }
        return res;
    }
}